src/node.rs

```diff
@@ -130,6 +130,29 @@ impl<T> Node<T> {
     }
 
     ///
+    /// Exchange the attached data through a new one
```

Comment:
This is confusing wording to me.

How does this sound to you:
"Swaps the value of this `Node`'s data and the data provided."

I'm still not 100% happy with that wording either tbh, but what do you think?